src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py

```diff
@@ -89,17 +89,26 @@ def delete_job(self, jobname, jobid):
             The name of the job
         jobid: str
             The job id on JES
+        modify_version: str
+            Default ("2.0") specifies that the request is to be processed synchronously. For asynchronous processing - change the value to "1.0"
 
         Returns
         -------
         response_json
             A JSON containing the result of the request execution
         """
+        if modify_version not in ("1.0", "2.0"):
+            raise ValueError('Accepted values: "1.0" or "2.0".')
+
         custom_args = self._create_custom_request_arguments()
         job_url = "{}/{}".format(jobname, jobid)
         request_url = "{}{}".format(self.request_endpoint, job_url)
         custom_args["url"] = request_url
-        response_json = self.request_handler.perform_request("DELETE", custom_args, expected_code=[202])
+        custom_args["headers"]["X-IBM-Job-Modify-Version"] = "2.0"
```

Comment:
You already verify on line 100 that `modify_version` is `1.0` or `2.0`. You can just assign the value directly to the custom argument.

Reply:
Assigned the value of modify_version to custom_args["headers"]["X-IBM-Job-Modify-Version"]